### ISSUE TITLE
Red and Green may be swapped in viz

### ISSUE THREAD

**Issue by LinearZoetrope** [SCAII/SCAII#55]
This is odd, I don't know why I didn't notice it before, I think I just assumed it was a weird choice I made, but I inspected my own values both in the packets and in the engine and they're correct:

![image](https://user-images.githubusercontent.com/3078382/35721300-246cb7ca-07a6-11e8-9000-92277dbb42cf.png)

This is the tower demo, and the reds and greens are swapped. I verified every step of the way from making the object to sending the message and somehow {r: 0, g: 255, b: 0} is displaying red and vice versa. I think this is a recent change maybe? I don't recall noticing it before.

The thing that tipped me off was remembering green was the "good" side, and the agent was red.

**Comment by LinearZoetrope** [comment]
I figured it out, the "triangle" is always rendered red. Is there any way you can fix this?

@jedirv

**Comment by jedirv** [comment]
Sorry didn't get to this over the weekend.  First on my list this morning.

**Comment by jedirv** [comment]
We may have discussed this long ago, but I found what I thought was another bug where I was dividing the A channel of RGBA by 255 to get a value to apply.  (the value ultimately needs to be 0- 1.0 value.  But then I looked at the message format for that:

message Color {
	required uint32 r = 1;
	required uint32 g = 2;
	required uint32 b = 3;
	required uint32 a = 4;
}
and see that a is a uint rather than a double, so it is the right thing to do to divide by 255 as long as the opacity you are asking for is in the 1-255 range.  Just mentioning this to make sure we're on the same page.
